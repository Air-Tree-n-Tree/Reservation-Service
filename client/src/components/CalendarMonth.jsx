import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';

const CalendarMonth = ({ month }) => (
  <div>
    <div>{moment(month).format('MMMM')}</div>
  </div>
);

CalendarMonth.propTypes = {
  month: PropTypes.string.isRequired,
};

const mapStateToProps = ({ availability }) => ({
  minNights: availability.minNights,
  maxNights: availability.maxNights,
});

export default connect(mapStateToProps)(CalendarMonth);
