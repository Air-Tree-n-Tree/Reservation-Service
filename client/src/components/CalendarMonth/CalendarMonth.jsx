import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import moment from 'moment';

import Day from '../CalendarDay/CalendarDay';
import classes from './CalendarMonth.module.css';

export const CalendarMonth = ({ month, reservedDays }) => {
  const startOfMonthDay = moment(month).day();
  const startOfMonthDaysSince2000 = moment(month).diff(moment('2000-01-01'), 'days');
  return (
    <div className={classes.calendarContainer}>
      <div className={classes.monthLabel}>{moment(month).format('MMMM')}</div>
      <div className={classes.calendar}>
        { ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'].map((dayName) => (
          <div key={dayName}>{ dayName }</div>
        )) }

        {
          startOfMonthDay > 0
            ? <div style={{ gridColumn: `1 / ${startOfMonthDay + 1}` }} />
            : null
        }

        { reservedDays.map((dayStatus, dayOfMonth) => {
          const day = startOfMonthDaysSince2000 + dayOfMonth;
          return (
            <Day
              key={day}
              day={day}
              dayOfMonth={dayOfMonth}
              status={dayStatus}
            />
          );
        })}
      </div>
    </div>
  );
};

CalendarMonth.propTypes = {
  month: PropTypes.string.isRequired,
  reservedDays: PropTypes.arrayOf(PropTypes.string).isRequired,
};

const mapStateToProps = ({ reservedDays, startingDay }, { month }) => {
  const startingIndex = moment(month).diff(moment('2000-01-01'), 'days') - startingDay;
  const endingIndex = startingIndex + moment(month).daysInMonth();
  return {
    startingIndex,
    reservedDays: reservedDays.slice(startingIndex, endingIndex),
  };
};

export default connect(mapStateToProps)(CalendarMonth);
