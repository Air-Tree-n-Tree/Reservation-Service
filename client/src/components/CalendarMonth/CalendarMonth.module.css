.calendar {
  display: grid;
  grid-template-columns: repeat(7, 50px);
  grid-template-rows: 30px;
  grid-auto-rows: 50px;
  place-items: center;
}
