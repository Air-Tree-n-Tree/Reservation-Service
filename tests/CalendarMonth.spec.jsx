import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import CalendarMonth from '../client/src/components/CalendarMonth/CalendarMonth';

describe('Calender Month component', () => {
  let calendarMonth;
  beforeEach(() => {
    const reservedDays = [];
    reservedDays.length = moment('2020-03').daysInMonth();
    reservedDays.fill('available');
    calendarMonth = shallow(
      <CalendarMonth
        month="2020-03"
        reservedDays={reservedDays}
        position={0}
      />,
    );
  });
  it('should render correctly', () => {
    expect(calendarMonth).toMatchSnapshot();
  });
});
