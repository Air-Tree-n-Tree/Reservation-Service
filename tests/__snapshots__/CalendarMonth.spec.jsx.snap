// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calender Month component should render correctly 1`] = `
<div
  className="calendarContainer"
  style={
    Object {
      "transform": "translateX(0%)",
    }
  }
>
  <div
    className="monthLabel"
  >
    March
  </div>
  <div
    className="calendar"
  >
    <div
      key="Su"
    >
      Su
    </div>
    <div
      key="Mo"
    >
      Mo
    </div>
    <div
      key="Tu"
    >
      Tu
    </div>
    <div
      key="We"
    >
      We
    </div>
    <div
      key="Th"
    >
      Th
    </div>
    <div
      key="Fr"
    >
      Fr
    </div>
    <div
      key="Sa"
    >
      Sa
    </div>
    <Connect(Day)
      day={7365}
      dayOfMonth={0}
      key="7365"
      status="available"
    />
    <Connect(Day)
      day={7366}
      dayOfMonth={1}
      key="7366"
      status="available"
    />
    <Connect(Day)
      day={7367}
      dayOfMonth={2}
      key="7367"
      status="available"
    />
    <Connect(Day)
      day={7368}
      dayOfMonth={3}
      key="7368"
      status="available"
    />
    <Connect(Day)
      day={7369}
      dayOfMonth={4}
      key="7369"
      status="available"
    />
    <Connect(Day)
      day={7370}
      dayOfMonth={5}
      key="7370"
      status="available"
    />
    <Connect(Day)
      day={7371}
      dayOfMonth={6}
      key="7371"
      status="available"
    />
    <Connect(Day)
      day={7372}
      dayOfMonth={7}
      key="7372"
      status="available"
    />
    <Connect(Day)
      day={7373}
      dayOfMonth={8}
      key="7373"
      status="available"
    />
    <Connect(Day)
      day={7374}
      dayOfMonth={9}
      key="7374"
      status="available"
    />
    <Connect(Day)
      day={7375}
      dayOfMonth={10}
      key="7375"
      status="available"
    />
    <Connect(Day)
      day={7376}
      dayOfMonth={11}
      key="7376"
      status="available"
    />
    <Connect(Day)
      day={7377}
      dayOfMonth={12}
      key="7377"
      status="available"
    />
    <Connect(Day)
      day={7378}
      dayOfMonth={13}
      key="7378"
      status="available"
    />
    <Connect(Day)
      day={7379}
      dayOfMonth={14}
      key="7379"
      status="available"
    />
    <Connect(Day)
      day={7380}
      dayOfMonth={15}
      key="7380"
      status="available"
    />
    <Connect(Day)
      day={7381}
      dayOfMonth={16}
      key="7381"
      status="available"
    />
    <Connect(Day)
      day={7382}
      dayOfMonth={17}
      key="7382"
      status="available"
    />
    <Connect(Day)
      day={7383}
      dayOfMonth={18}
      key="7383"
      status="available"
    />
    <Connect(Day)
      day={7384}
      dayOfMonth={19}
      key="7384"
      status="available"
    />
    <Connect(Day)
      day={7385}
      dayOfMonth={20}
      key="7385"
      status="available"
    />
    <Connect(Day)
      day={7386}
      dayOfMonth={21}
      key="7386"
      status="available"
    />
    <Connect(Day)
      day={7387}
      dayOfMonth={22}
      key="7387"
      status="available"
    />
    <Connect(Day)
      day={7388}
      dayOfMonth={23}
      key="7388"
      status="available"
    />
    <Connect(Day)
      day={7389}
      dayOfMonth={24}
      key="7389"
      status="available"
    />
    <Connect(Day)
      day={7390}
      dayOfMonth={25}
      key="7390"
      status="available"
    />
    <Connect(Day)
      day={7391}
      dayOfMonth={26}
      key="7391"
      status="available"
    />
    <Connect(Day)
      day={7392}
      dayOfMonth={27}
      key="7392"
      status="available"
    />
    <Connect(Day)
      day={7393}
      dayOfMonth={28}
      key="7393"
      status="available"
    />
    <Connect(Day)
      day={7394}
      dayOfMonth={29}
      key="7394"
      status="available"
    />
    <Connect(Day)
      day={7395}
      dayOfMonth={30}
      key="7395"
      status="available"
    />
  </div>
</div>
`;
